src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -199,21 +218,70 @@ public DRSAccessURL getAccessUrlForObjectId(
 
     DrsId drsId = drsIdService.fromObjectId(objectId);
     Snapshot snapshot = snapshotService.retrieve(UUID.fromString(drsId.getSnapshotId()));
-    GoogleProjectResource projectResource = snapshot.getProjectResource();
 
-    DRSAccessMethod matchingAccessMethod =
-        getAccessMethodMatchingAccessId(accessId, drsObject)
-            .orElseThrow(
-                () ->
-                    new IllegalArgumentException(
-                        "No matching access ID " + accessId + " was found on object " + objectId));
+    BillingProfileModel billingProfileModel =
+        snapshot
+            .getFirstSnapshotSource()
+            .getDataset()
+            .getDatasetSummary()
+            .getDefaultBillingProfile();
+
+    Supplier<IllegalArgumentException> illegalArgumentExceptionSupplier =
+        () -> {
+          throw new IllegalArgumentException("No matching access ID was found for object");
+        };
+
+    CloudPlatformWrapper wrapper = CloudPlatformWrapper.of(billingProfileModel.getCloudPlatform());
+    if (wrapper.isGcp()) {
+      DRSAccessMethod matchingAccessMethod =
+          getAccessMethodMatchingAccessId(accessId, drsObject, TypeEnum.GS)
+              .orElseThrow(illegalArgumentExceptionSupplier);
+      return signGoogleUrl(snapshot, matchingAccessMethod);
+    } else if (wrapper.isAzure()) {
+      getAccessMethodMatchingAccessId(accessId, drsObject, TypeEnum.HTTPS)
+          .orElseThrow(illegalArgumentExceptionSupplier);
+      try {
+        FSItem fsItem =
+            fileService.lookupSnapshotFSItem(
+                snapshotService.retrieveAvailableSnapshotProject(snapshot.getId()),
+                drsId.getFsObjectId(),
+                1);
+        return signAzureUrl(billingProfileModel, fsItem);
```

Comment:
I think you could make `signAzureUrl` take a cloudPath rather than fsItem, in which case, you could actually use the result of  `getAccessMethodMatchingAccessId` on line 241 (to avoid multiple calls)

Reply:
Err...is the reason we're not so we can get the storage account resource?

Reply:
Eh, that might be a bigger change. TODO, we should do some catching on storage account lookups but I think we're ok for now